### ISSUE TITLE
Allow for notifications across all pages in Site Kit to display notifications contextually for the modules and setup

### ISSUE THREAD

**Issue by bjackson27** [google/site-kit-wp#188]
Allow for notifications on the module pages. So that AdSense notifications would be served along AdSense and that Analytics would be served on the analytics page. The notifications would still appear on the main dashboard.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Product-specific notifications are displayed both on the main plugin dashboard and in the product sub-report. 
* If there is more than one notification from the same product, on the main plugin dashboard they get grouped, in order not to take up too much screen real estate. 
* If the user dismisses the notification from either place in the plugin, it should also disappear from the other. 
* Plugin-wide notifications (e.g. increase in visitors etc) appear only on the main dashboard. 

## Implementation Brief

* Implement a "site-wise" notifications endpoint to compliment https://github.com/google/site-kit-wp/blob/93ca9a922f4091f6ddf25b970133c8639542f723/includes/Core/REST_API/REST_Routes.php#L545 (this will require a list of notifications that apply to the whole site that it should return, as well as—possibly—module-specific notifications I suppose)

## Changelog entry

*

**Comment by tofumatt** [comment]
I'm a bit unclear as to where we fetch "plugin-wide" notifications from. I don't see a separate  endpoint for notifications that apply to the whole site. We would need to add this endpoint (as right now we fetch notifications based on module name: https://github.com/google/site-kit-wp/blob/93ca9a922f4091f6ddf25b970133c8639542f723/includes/Core/REST_API/REST_Routes.php#L545)

The example given "increase in visitors" seems to be an analytics-related notification to me, and I'm guessing comes from the analytics notifications API (I saw it when I connected analytics to a site recently). Do we have a list of notifications considered "site-wide" and from which service they come from right now? I think we'd need those to sort out this implementation brief and to make the AC clearer as to what notifications we want to include only on the "main" Dashboard.

I've also not encountered more than one notification at a time—is grouping notifications something already implemented (that I've not encountered) or is there a particular UX mockup/etc. we should work from for this?
